include/swift/SILOptimizer/Utils/Differentiation/ADContext.h

```diff
@@ -0,0 +1,349 @@
+//===--- ADContext.h - Context for Differentiation ----------*- C++ -*---===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// SWIFT_ENABLE_TENSORFLOW
+//
+// Per-module contextual information for the Differentiation pass.
+//===----------------------------------------------------------------------===//
```

Comment:
(minor, feel free to address later)

Empty line between file header comment and code please.
```suggestion
//===----------------------------------------------------------------------===//

```